apps/web/src/components/forms/password.tsx

```diff
@@ -18,12 +18,40 @@ import { useToast } from '@documenso/ui/primitives/use-toast';
 
 import { FormErrorMessage } from '../form/form-error-message';
 
+const commonPasswords = [
+  'password',
+  '12345678',
+  'password123',
+  'admin',
+  'qwerty',
+  'poiuyt',
+  'qwerty123',
+  'qwerty@123',
+];
+
 export const ZPasswordFormSchema = z
   .object({
-    currentPassword: z.string().min(6).max(72),
-    password: z.string().min(6).max(72),
-    repeatedPassword: z.string().min(6).max(72),
+    currentPassword: z.string().min(8).max(72),
+    password: z
+      .string()
+      .regex(new RegExp('.*[A-Z].*'), { message: 'One uppercase character' })
+      .regex(new RegExp('.*[a-z].*'), { message: 'One lowercase character' })
+      .regex(new RegExp('.*\\d.*'), { message: 'One number' })
+      .regex(new RegExp('.*[`~<>?,./!@#$%^&*()\\-_+="\'|{}\\[\\];:\\\\].*'), {
+        message: 'One special character is required',
+      })
+      .min(8, { message: 'Must be at least 8 characters in length' })
+      .max(72, { message: 'Cannot be more than 72 characters in length' }),
```

Comment:
Extract this out to something like `ZPasswordSchema` which then can be reused between `ZPasswordFormSchema` and `ZSignUpFormSchema`

Reply:
Gotcha, will update it :)